profile/service.go

```diff
@@ -132,3 +132,14 @@ func parseIsAgeVerifiedValue(byteValue []byte) (result *bool, err error) {
 
 	return
 }
+
+func handleUnsuccessfulShare(parsedResponse profileDO) error {
+	if parsedResponse.ErrorDetails != nil {
+		return yotierror.DetailedSharingFailureError{
+			Code:        parsedResponse.ErrorDetails.ErrorCode,
```

Comment:
Is it possible that there may be non nil ErrorDetails but nil 
`parsedResponse.ErrorDetails.ErrorCode`
or 
`parsedResponse.ErrorDetails.Description`
?

Reply:
Good question - I dont believe so, but def worth double checking. Ill take a look

Reply:
Yeah my understanding is that for the specific use case this is being added for these KVPs _will_ be present, but there _may possibly_ be other cases where there's no error_code at least. But I am not sure. I believe Paolo and/or Joe would know.

Reply:
I see the commit to update this - thanks - am wondering though, dotnet side we capture the whole of the Response content whenever `SharingOutcome != "SUCCESS"` to cover cases where the above conditions are not met. 

This provides visibility of the entire response Json _regardless of the presence/absence of particular key value pairs_. 

`ResponseContent = responseContent;`

https://github.com/getyoti/yoti-dotnet-sdk/blob/f9fda98dce74bf89e33744b718449f83db7bd1f5/src/Yoti.Auth/Exceptions/YotiProfileException.cs

Perhaps you want to do the same here?